services/system/HttpServer/src/HttpServer.cpp

```diff
@@ -211,46 +213,41 @@ namespace SystemService
 
    extern "C" [[clang::export_name("serve")]] void serve()
    {
-      auto act = getCurrentAction();
+      auto act   = getCurrentAction();
+      auto iface = [&](AccountNumber server)
+      { return psibase::Actor<ServerInterface>(act.service, server); };
+
       // TODO: use a view
       auto [sock, req] = psio::from_frac<std::tuple<std::int32_t, HttpRequest>>(act.rawData);
 
-      std::string serviceName;
-
-      // Path reserved across all subdomains
-      if (req.target.starts_with("/common"))
-         serviceName = "common-api";
-
-      // subdomain
-      else if (isSubdomain(req))
-         serviceName.assign(req.host.begin(), req.host.end() - req.rootHost.size() - 1);
-
-      // root domain
-      else
-         serviceName = "homepage";
-
-      auto service = AccountNumber(serviceName);
-      auto reg     = kvGet<RegisteredServiceRow>(registeredServiceKey(act.service, service));
-      if (reg)
-         service = reg->server;
-      else
-         service = "sites"_a;
-
-      currentRequest = {.socket = sock, .owner = service};
+      // First we check the `sites` server
+      auto server    = "sites"_a;
+      currentRequest = {.socket = sock, .owner = server};
+      auto result    = iface(server).serveSys(req, std::optional{sock});
 
-      psibase::Actor<ServerInterface> iface(act.service, service);
-      auto                            result = iface.serveSys(req, std::optional{sock});
+      // If not found, we check the registered server
+      if (!result)
```

Comment:
This needs to check `currentRequest` as well. (Either that or don't set currentRequest and don't pass the socket to `sites`)

Reply:
Which is preferred?

Reply:
I think I prefer checking currentRequest. If I understand correctly, if I don't send the socket to sites, it wouldn't be able to facilitate wss connections or otherwise defer replies, etc. Not needed for now, but possibly in the future.

Reply:
Done